API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java

```diff
@@ -26,31 +29,31 @@ public QueryController(QueryService queryService) {
     }
 
     @PostMapping(value = "/queries")
-    public ResponseEntity query(@RequestParam String query) {
+    public Query query(@RequestParam String query) {
         try {
             logger.info("POST /queries: {}", query);
-            return ResponseEntity.status(200).body(gson.toJson(queryService.createQuery(query)));
+            return queryService.createQuery(query);
         } catch (Exception e) {
             throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "", e);
         }
     }
 
     @GetMapping(value = "/queries")
-    public ResponseEntity queries() {
+    public List<Query> queries() {
         try {
             logger.info("GET /queries");
-            return ResponseEntity.status(200).body(gson.toJson(queryService.getAllQueries()));
+            return queryService.getAllQueries();
         } catch (Exception e) {
             throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "", e);
         }
     }
 
     @GetMapping(value = "/queries/{queryID}")
-    public ResponseEntity singleQuery(@PathVariable("queryID") String queryID) {
+    public Optional<Query> singleQuery(@PathVariable("queryID") String queryID) {
         try {
             logger.info("GET /queries/" + queryID);
             logger.info(queryService.getQueryByID(queryID).toString());
```

Comment:
You are calling `queryService.getQueryByID(queryID)` twice, here and below.

Reply:
There is no reason to log the entire query, so I'd remove this. If you want to keep it, put it into debug mode (debugging is the only valid reason for logging this, imo)